oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderJwkSupport.java

```diff
@@ -111,6 +111,10 @@ public Jwt decode(String token) throws JwtException {
 		throw new JwtException("Unsupported algorithm of " + jwt.getHeader().getAlgorithm());
 	}
 
+	public String getJwkSetUrl() {
```

Comment:
This is not used anywhere. We should remove it.

Reply:
@jgrandja please consult one of the unit tests, which confirms that the configuration took effect. The two alternatives are to use reflection to confirm this setting or to perform some heavier configuration to mock this url endpoint in addition to the openid-configuration endpoint already mocked.

I'm okay with the first option, though not precisely ideal, though the second option I believe increases the scope of the test too much.

What would you do in this situation?

Reply:
Also, I believe this situation would be simplified if we could expose RestOperations as configurable here as well as in NimbusJwtDecoderJwkSupport. Then, this test could provide a mock RestOperations and alleviate the need to mock an underlying server.

That being the case, I'm also okay with leaving the unit test in a suboptimal state (calling reflectively a private method) in anticipation of the class under test being able to more easily configure mock endpoints.

Reply:
I must be missing something? I don't see any references to `getJwkSetUrl()` after searching for usages within IDE.

> I believe this situation would be simplified if we could expose RestOperations as configurable here as well as in NimbusJwtDecoderJwkSupport

This is coming in [#5603](https://github.com/spring-projects/spring-security/issues/5603)